lib/mongoid/lazy_migration/document.rb

```diff
@@ -63,8 +63,7 @@ def try_lock_migration
     selector = atomic_selector.merge('migration_state' => { "$in" => [nil, 'pending'] })
     changes  = { "$set" => { 'migration_state' => 'processing' }}
     safety   = { :safe => true }
-
-    self.class.with(safety).where(selector).query.update(changes)['updatedExisting']
+    self.class.with(safety).where(selector).update(changes).successful?
```

Comment:
i'm not 100% sure the "true" we are returning now from here is the same as the one that was before. Because mongodb doesn't return the updatedExisting flag anymore.